### PR TITLE
Add upgrade-downgrade tests for next releases

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR adds the next-release counterparts of the current upgrade-downgrade tests. Currently, we have tests that test the upgrade process from the previous release to the current one. For example on `main` the tests run against release-14.0. On release-14.0 they run against 13.0.2 and so on.
But it is essential to also run the tests against the next release. For example, if we introduce a change in release-13.0 which would prevent it from upgrading to 14.0.1, we won't catch it at all until a new patch release is done, in which case it would fail in a PR based against release-14.0

To also catch these scenarios, we have chosen to add upgrade-downgrade tests to test against the next release apart from just the previous release. For `main`, the tests will just be skipped. For release-14.0, `main` will be used, for release-13.0, `14.0.1` will be used and so on.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: tools/get_next_release.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to build and copy the Angular 2 based vtctld UI
+# into the release folder (app) for checkin. Prior to running this script,
+# bootstrap.sh and bootstrap_web.sh should already have been run.
```
Comment (frouioui) [inline tools/get_next_release.sh:19]: nit: we can remove this

Comment (GuptaManan100) [reply tools/get_next_release.sh:19]: Oh yes, that seems wrong!

**Comment by frouioui** [comment]
How about reusing the already existing workflows and changing them to run using a dynamic matrix that gets created in the `get_next_release`/`get_previous_release` job. The next job, `upgrade_downgrade_test`, can run using this matrix.

A similar use case is used in `release-12.0`: https://github.com/vitessio/vitess/blob/e00c46a6380d21c318ec19737fa16d438326e1fa/.github/workflows/cluster_endtoend_upgrade.yml#L43-L45

**Comment by GuptaManan100** [comment]
@frouioui I thought of doing that, but then I thought that it would serialize the workflows and increase its total running time a lot. I think separate workflows is fine since each of them anyways runs for like 40-50 minutes.

File: tools/get_next_release.sh (hunk 1 above)
Comment (GuptaManan100) [reply tools/get_next_release.sh]: Fixed

**Comment by GuptaManan100** [comment]
It can be seen in the backport PRs to release-14.0 https://github.com/vitessio/vitess/pull/11034 and release-13.0 https://github.com/vitessio/vitess/pull/11040, the tests are working as expected

**Comment by frouioui** [comment]
> @frouioui I thought of doing that, but then I thought that it would serialize the workflows and increase its total running time a lot. I think separate workflows is fine since each of them anyways runs for like 40-50 minutes.

@GuptaManan100, the jobs of a matrix can be parallelized: https://docs.github.com/en/actions/using-workflows/reusing-workflows. The PR is looking good to me anyway, feel free to merge or use the matrix thingy.

**Review by frouioui — APPROVED** [review]

**Comment by GuptaManan100** [comment]
@frouioui I agree the matrix approach would be better because there will be fewer workflows to maintain, but since the workflows already work and backports are ready, I'm going to merge this PR. We can refactor the workflows later when we want to add more tests.

**Pull request merged**